custom_components/opensprinkler/switch.py

```diff
@@ -90,15 +89,15 @@ def _get_state(self) -> str:
         """Retrieve latest state."""
         return bool(self._controller.enabled)
 
-    def turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs):
```

Comment:
I am thinking we may also need to update the service `run` and `stop` to async

Reply:
Yeah I wasn’t sure what was needed for that so I left them. Want me to make the changes or you want to push a commit to this branch?

Reply:
If it still works maybe we can do it separately in another PR

Reply:
But if you want to do it then go ahead,

Reply:
I can just try changing them to async and see what happens.

Reply:
Do you know how to pass arguments to this kind of invocation?

`hass.async_add_executor_job(hub.update)`

Reply:
I believe its: 

`hass.async_add_executor_job(hub.update, arg1, arg2, ...)`

Reply:
OK, I think that worked. If you want to try out the branch and see how it goes for you that would be great.  Didn't test the service calls extensively but I believe the syntax is correct.